packages/protocol/contracts/ERC1724/base/ZkAssetMintableBase.sol

```diff
@@ -49,21 +49,22 @@ contract ZkAssetMintableBase is ZkAssetOwnableBase {
     }
 
     /**
-    * @dev Executes a basic unilateral, confidential transfer of AZTEC notes adapted for use with
-    * a mintable ZkAsset.
-    *
+    * @dev Executes a basic unilateral, confidential transfer of AZTEC notes
     * Will submit _proofData to the validateProof() function of the Cryptography Engine.
     *
-    * If public value is being transferred out of the ACE, and the minted value is greater than
-    * ACE's token balance, then tokens will be minted from the linked ERC20 token using supplementTokens()
-    *
     * Upon successfull verification, it will update note registry state - creating output notes and
     * destroying input notes.
     *
-    * @param _proofData bytes variable outputted from proof construction
-    * @param _signatures ECDSA signatures over all input notes involved in the confidentialTransfer()
+    * @param _proofId - id of proof to be validated. Needs to be a balanced proof.
+    * @param _proofData - bytes variable outputted from a proof verification contract, representing
+    * transfer instructions for the ACE
+    * @param _signatures - array of the ECDSA signatures over all inputNotes
     */
-    function confidentialTransfer(bytes memory _proofData, bytes memory _signatures) public {
+    function confidentialTransfer(uint24 _proofId, bytes memory _proofData, bytes memory _signatures) public {
+        // Check that it's a balanced proof
+        (, uint8 category, ) = _proofId.getProofComponents();
+
+        require(category == uint8(ProofCategory.BALANCED), "this is not a balanced proof");
```

Comment:
Good point, `confidentialTransfer()` in the `ZkAssetMintableBase.sol` should perform a check/some form of permissioning that the type of proof passed is supported by the asset